src/java/org/apache/cassandra/tools/SSTableMetadataViewer.java

```diff
@@ -20,32 +20,58 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.PrintStream;
-import java.util.EnumSet;
-import java.util.Map;
+import java.util.*;
+import java.util.stream.Collectors;
 
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.google.common.collect.Sets;
 import org.apache.cassandra.io.sstable.Descriptor;
 import org.apache.cassandra.io.sstable.metadata.*;
+import org.apache.commons.cli.*;
 
 /**
  * Shows the contents of sstable metadata
  */
 public class SSTableMetadataViewer
 {
+    private static final String ANCESTOR_OPTION = "a";
+    private static final Options options = new Options();
+    private static CommandLine cmd;
+
+    static
+    {
+        Option optKey = new Option(ANCESTOR_OPTION, true, "Ancestor generation");
+        // Number of times -a <ancestor> can be passed on the command line.
+        optKey.setArgs(500);
+        options.addOption(optKey);
+    }
+
     /**
      * @param args a list of sstables whose metadata we're interested in
      */
     public static void main(String[] args) throws IOException
     {
+        String usage = "Usage: sstablemetadata <sstable filenames> [-a ancestor [-a ancestor [...]]";
         PrintStream out = System.out;
-        if (args.length == 0)
+
+        CommandLineParser parser = new PosixParser();
+        try
+        {
+            cmd = parser.parse(options, args);
+        }
+        catch (ParseException e1)
         {
-            out.println("Usage: sstablemetadata <sstable filenames>");
+            System.err.println(e1.getMessage());
+            System.err.println(usage);
             System.exit(1);
         }
 
         Util.initDatabaseDescriptor();
 
-        for (String fname : args)
+        String[] ancestorsMaybeNull = cmd.getOptionValues(ANCESTOR_OPTION);
+        Set<Integer> ancestors = Arrays.stream(ancestorsMaybeNull != null ? ancestorsMaybeNull : new String[0]).map(Integer::parseInt).collect(Collectors.toSet());
+        boolean filterAncestors = !ancestors.isEmpty();
```

Comment:
i can imagine that this would be useful, but that should be done by a new option like `-e`. `boolean filterAncestors = !ancestors.isEmpty();` maintains prior behavior where if you don't specify `-a` options, no filtering will be applied